test/route_end_to_end_connectivity_test.cc

```diff
@@ -0,0 +1,291 @@
+    lane_2_0_1 = road_network_->road_geometry()->ById().GetLane(kLane_2_0_1);
+    lane_2_0_m1 = road_network_->road_geometry()->ById().GetLane(kLane_2_0_m1);
+
+    ASSERT_NE(nullptr, lane_1_0_1);
+    ASSERT_NE(nullptr, lane_1_0_m1);
+    ASSERT_NE(nullptr, lane_2_0_1);
+    ASSERT_NE(nullptr, lane_2_0_m1);
+  }
+};
+
+class SinglePhaseTest : public MultipleStraightSegmentsTest {};
+
+TEST_F(SinglePhaseTest, SinglePhaseWithOneLaneIsConnectedEndToEnd) {
+  const api::RoadPosition start_position{lane_1_0_1, api::LanePosition{0., 0., 0.}};
+  const api::RoadPosition end_position{lane_1_0_1, api::LanePosition{kLaneLength, 0., 0.}};
+  const api::LaneSRange lane_s_range(kLane_1_0_1, api::SRange(0, kLaneLength));
+  const routing::Phase phase(0 /*index*/, kLinearTolerance, {start_position}, {end_position}, {lane_s_range},
+                             road_network_.get());
+  const routing::Route dut({phase}, road_network_.get());
+
+  const std::vector<std::string> errors = dut.ValidateEndToEndConnectivity();
+
+  ASSERT_TRUE(errors.empty()) << SerializeErrors(errors);
+}
+
+TEST_F(SinglePhaseTest, SinglePhaseWithTwoLanesIsConnectedEndToEnd) {
+  const api::RoadPosition start_position{lane_1_0_1, api::LanePosition{0., 0., 0.}};
+  const api::RoadPosition end_position{lane_1_0_1, api::LanePosition{kLaneLength, 0., 0.}};
+  const api::LaneSRange lane_s_range_1(kLane_1_0_1, api::SRange(0, kLaneLength));
+  const api::LaneSRange lane_s_range_m1(kLane_1_0_m1, api::SRange(0, kLaneLength));
+  const routing::Phase phase(0 /*index*/, kLinearTolerance, {start_position}, {end_position},
+                             {lane_s_range_m1, lane_s_range_1}, road_network_.get());
+  const routing::Route dut({phase}, road_network_.get());
+
+  const std::vector<std::string> errors = dut.ValidateEndToEndConnectivity();
+
+  ASSERT_TRUE(errors.empty()) << SerializeErrors(errors);
+}
+
+TEST_F(SinglePhaseTest, SinglePhaseWithTwoLanesAndStartPositionsIsNotConnectedEndToEnd) {
+  const api::RoadPosition start_position_1{lane_1_0_1, api::LanePosition{0., 0., 0.}};
+  const api::RoadPosition start_position_m1{lane_1_0_m1, api::LanePosition{0., 0., 0.}};
+  const api::RoadPosition end_position{lane_1_0_1, api::LanePosition{kLaneLength, 0., 0.}};
+  const api::LaneSRange lane_s_range_1(kLane_1_0_1, api::SRange(0, kLaneLength));
+  const api::LaneSRange lane_s_range_m1(kLane_1_0_m1, api::SRange(0, kLaneLength));
+  const routing::Phase phase(0 /*index*/, kLinearTolerance, {start_position_1, start_position_m1}, {end_position},
+                             {lane_s_range_m1, lane_s_range_1}, road_network_.get());
+  const routing::Route dut({phase}, road_network_.get());
+
+  const std::vector<std::string> errors = dut.ValidateEndToEndConnectivity();
+
+  ASSERT_FALSE(errors.empty()) << SerializeErrors(errors);
+}
+
+TEST_F(SinglePhaseTest, SinglePhaseWithTwoLanesAndEndPositionsIsNotConnectedEndToEnd) {
```

Comment:
Ditto. Add a comment explaining why having end positions results in the route not being connected end-to-end.

Reply:
[5415ba7](https://github.com/maliput/maliput_integration_tests/pull/81/commits/5415ba7b4a80d037e1a4fe16999f64b6bfd528aa)